File: src/app-project/ios/proshake/glitter_interface/getInset.swift
//
//  getInset.swift
//  proshake
//
//  Created by Jianzhi.wang on 2023/7/9.
//

import Foundation
import Glitter_IOS
import UIKit

class GetInset{
    public static func createShareInterface(){
        GlitterActivity.sharedInterFace.append(JavaScriptInterFace(functionName: "getTopInset", function: {
            request in
            request.responseValue["data"]=request.glitterAct.safeAreaInsetsTop
            request.finish()
        }))
        //Get top safearea distance
        GlitterActivity.sharedInterFace.append(JavaScriptInterFace(functionName: "getBottomInset", function: {
            request in
            request.responseValue["data"]=request.glitterAct.safeAreaInsetsBottom
            request.finish()
        }))
    }
}

extension UIViewController {
    func presentFullScreen(_ viewControllerToPresent: UIViewController, animated flag: Bool, completion: (() -> Void)? = nil) {
        viewControllerToPresent.modalPresentationStyle = .fullScreen
        present(viewControllerToPresent, animated: flag, completion: completion)
    }

    var topbarHeight: CGFloat {
        return safeAreaInsetsTop + (navigationController?.navigationBar.frame.height ?? 0.0)
    }
    var toolbarHeight: CGFloat {
        return  safeAreaInsetsBottom + (navigationController?.toolbar.frame.height ?? 0.0)
    }
    
    var safeAreaInsetsTop: CGFloat {
        return  (ViewController.vc?.view.safeAreaInsets.top ?? 0.0)
    }
    
    
    var safeAreaInsetsBottom: CGFloat {
        return  (ViewController.vc?.view.safeAreaInsets.bottom ?? 0.0)
    }
    
}

File: src/app-project/android/app/src/main/java/com/ncdesign/kenda/glitter_interface/GetInset.kt
package com.ncdesign.kenda.glitter_interface

import com.jianzhi.glitter.GlitterActivity
import com.jianzhi.glitter.JavaScriptInterFace

class GetInset {
    companion object{
        fun createShareInterface(){
            GlitterActivity.addJavacScriptInterFace(JavaScriptInterFace("getTopInset") {
                it.responseValue["data"]=0
                it.fin()
            })
            GlitterActivity.addJavacScriptInterFace(JavaScriptInterFace("getBottomInset") {
                it.responseValue["data"]=0
                it.fin()
            })
        }
    }
}